src/main/java/com/synopsys/integration/detect/util/filter/DetectOverrideableFilter.java

```diff
@@ -24,16 +24,21 @@
 
 import java.util.Set;
 
+import com.synopsys.integration.detect.configuration.help.DetectArgumentState;
 import com.synopsys.integration.util.ExcludedIncludedWildcardFilter;
 import com.synopsys.integration.util.TokenizerUtils;
 
 public class DetectOverrideableFilter extends ExcludedIncludedWildcardFilter implements DetectFilter {
-    public DetectOverrideableFilter(final String toExclude, final String toInclude) {
+    public static DetectOverrideableFilter createArgumentValueFilter(DetectArgumentState detectArgumentState) {
```

Comment:
I don't really understand this change. Why couple the filter to the argument state?

Reply:
There were a number of places in the source where we were doing

`new DetectOverrideableFilter("", detectArgumentState.getParsedValue())`

This is simply a static constructor to replace that.